drivers/ethernet/eth_mcux.c

```diff
@@ -932,6 +951,30 @@ static void eth_mcux_ptp_isr(void *p)
 }
 #endif
 
+#ifdef CONFIG_SOC_SERIES_IMX_RT
+static void eth_mcux_dispacher_isr(void *p)
+{
+	struct device *dev = p;
+	struct eth_context *context = dev->driver_data;
+	u32_t EIR = ENET_GetInterruptStatus(ENET);
+	int irq_lock_key = irq_lock();
+
+	if (EIR & (kENET_RxBufferInterrupt | kENET_RxFrameInterrupt)) {
+		ENET_ReceiveIRQHandler(ENET, &context->enet_handle);
+	} else if (EIR & (kENET_TxBufferInterrupt | kENET_TxFrameInterrupt)) {
+		ENET_TransmitIRQHandler(ENET, &context->enet_handle);
+	} else if (EIR & ENET_EIR_MII_MASK) {
+		k_work_submit(&context->phy_work);
+		ENET_ClearInterruptStatus(ENET, kENET_MiiInterrupt);
+	} else if (EIR) {
+		ENET_ClearInterruptStatus(ENET, 0xFFFFFFFF);
+	}
+
+	irq_unlock(irq_lock_key);
+}
+#endif /* CONFIG_SOC_SERIES_IMX_RT */
+
+#ifdef CONFIG_SOC_SERIES_KINETIS_K6X
```

Comment:
Change this to `#ifdef ETH_IRQ_RX` and wrap the other isr's similarly with `#ifdef ETH_IRQ_TX` and `#ifdef ETH_IRQ_ERR_MISC`